common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/ExternalShuffleBlockResolver.java

```diff
@@ -374,6 +379,27 @@ public int removeBlocks(String appId, String execId, String[] blockIds) {
       .collect(Collectors.toMap(Pair::getKey, Pair::getValue));
   }
 
+  /**
+   * Diagnose the possible cause of the shuffle data corruption by verify the shuffle checksums
+   */
+  public Cause diagnoseShuffleBlockCorruption(
+      String appId,
+      String execId,
+      int shuffleId,
+      long mapId,
+      int reduceId,
+      long checksumByReader,
+      String algorithm) {
+    ExecutorShuffleInfo executor = executors.get(new AppExecId(appId, execId));
+    String fileName = "shuffle_" + shuffleId + "_" + mapId + "_0.checksum." + algorithm;
```

Comment:
After we pass the `algorithm` to the server, finding the checksum file is no longer an issue. cc @mridulm @otterc

Reply:
That is a cleaner solution, thanks !
Will also mean we can spread the checksums without needing to couple it with the shuffle file.

Reply:
nit: Btw, add a comment that this should be in sync with `IndexShuffleBlockResolver.getChecksumFile` ?